plugin/src/main/groovy/com/novoda/staticanalysis/internal/lint/LintConfigurator.groovy

```diff
@@ -28,16 +30,30 @@ class LintConfigurator implements Configurator {
 
     @Override
     void execute() {
-        project.extensions.findByType(StaticAnalysisExtension).ext."lintOptions" = { Closure config ->
-            if (!isAndroidProject(project)) {
-                return
+        project.extensions.findByType(StaticAnalysisExtension).ext.lintOptions = { Closure config ->
+            project.plugins.withId('com.android.application') {
+                configureWithVariants(config, filteredApplicationVariants)
             }
-            configureLint(config)
-            configureToolTask()
+            project.plugins.withId('com.android.library') {
+                configureWithVariants(config, filteredLibraryVariants)
+            }
+
+        }
+    }
+
+    private void configureWithVariants(Closure config, DomainObjectSet variants) {
+        configureLint(config)
+        if (hasFilter) {
```

Comment:
where `hasFilter` is coming from now? not sure how this is working now - if it's working even